### PR TITLE
feat: chart styles and loading state

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

- Loading state for new chart component
- Styles for the charts
- Chart header component

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

low risk

> What protocols, transaction types or contract interactions might be affected by this PR?

### Operations

The chart should show a loading state when its loading data.

When you scrub over the chart, the data should be displayed with the amount and date in the top left corner.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -165,6 +165,7 @@
     "p-memoize": "^7.1.1",
     "p-queue": "^8.0.1",
     "p-ratelimit": "^1.0.1",
+    "polished": "^4.3.1",
```
Comment (gomesalexandre) [inline package.json:168]: Don't think adding 2.93 MB is worth it for a single use-case https://packagephobia.com/result?p=polished
```typescript
import { lighten } from "@chakra-ui/theme-tools";
```

Though it's deprecated and could be removed anytime we update chakra-ui, but until then, we could use it

File: src/components/SimpleChart/ChartHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    <Heading as='h3' lineHeight={1}>
+      {toFiat(bnOrZero(value).toString())}
+    </Heading>
+  )
+}
+
+interface HeaderTimeDisplayProps {
+  time?: UTCTimestamp
+  /** Optional string to display when time is undefined */
+  timePlaceholder?: string
```
Comment (gomesalexandre) [inline src/components/SimpleChart/ChartHeader.tsx:56]: Do we need this? AFAIK if `time` is undefined, then sorting should be rugged and the chart should throw

File: src/components/SimpleChart/SimpleChart.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    (e: React.TouchEvent<HTMLDivElement>) => e.preventDefault(),
+    [],
+  )
+
+  const chartHeader = useMemo(() => {
+    return (
+      <ChartHeader
+        value={crosshairData?.value ?? lastPrice.value}
+        time={crosshairData?.time as UTCTimestamp}
+        timePlaceholder={formatHistoryDuration(interval)}
```
Comment (gomesalexandre) [inline src/components/SimpleChart/SimpleChart.tsx:253]: We need to `translate()` this

image

File: src/components/SimpleChart/ChartHeader.tsx (hunk 2 above)
Comment (reallybeard) [reply src/components/SimpleChart/ChartHeader.tsx:56]: Not whats its used for, when we get undefined that means they aren't scrubbing the chart.

File: src/components/SimpleChart/ChartHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import type { UTCTimestamp } from 'lightweight-charts'
+import type { ReactElement, ReactNode } from 'react'
+import { useTranslate } from 'react-polyglot'
+import styled from 'styled-components'
+import { Row } from 'components/Row/Row'
+import { useLocaleFormatter } from 'hooks/useLocaleFormatter/useLocaleFormatter'
+
+import { useHeaderDateFormatter } from './hooks'
+
+const ChartHeaderWrapper = (props: FlexProps) => (
```
Comment (gomesalexandre) [inline src/components/SimpleChart/ChartHeader.tsx:13]: We're not passing any prop in currently, let's remove the props?

File: src/components/SimpleChart/ChartHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  width: 70%;
+
+  * {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+`
+
+interface HeaderValueDisplayProps {
```
Comment (gomesalexandre) [inline src/components/SimpleChart/ChartHeader.tsx:33]: type plz

File: src/components/SimpleChart/ChartHeader.tsx (hunk 4 above)
Comment (reallybeard) [reply src/components/SimpleChart/ChartHeader.tsx:13]: we are passing children

Comment (reallybeard) [reply src/components/SimpleChart/ChartHeader.tsx:13]: But fair enough don't need to customize this

**Review by gomesalexandre — COMMENTED** [review]
Tested against develop:

- Selecting week liquidity chart will rug us and render the wrong series:

image

- Loading states are happy

https://github.com/shapeshift/web/assets/17035424/4ee085ae-cfd5-4670-8f1b-ce369678cded

**Review by gomesalexandre — APPROVED** [review]
All comments have been tackled, and confirmed the week visual change isn't a regression, get in

**Pull request merged**